src/nvim/buffer_defs.h

```diff
@@ -361,14 +361,36 @@ struct mapblock {
   sctx_T m_script_ctx;          // SCTX where map was defined
 };
 
-/*
- * Used for highlighting in the status line.
- */
+/// Used for highlighting in the status line.
+typedef struct stl_hlrec stl_hlrec_t;
```

Comment:
You removed the `stl_hlrec` in your Vim patch. Any compile issues?

Reply:
Well, Bram removed it, my PR included it. I looked at other definitions in `buffer_defs.h` and newer code (documented with `///`) tended to separate typedef & struct. I've used that style.